services/payments/secrets.go

```diff
@@ -39,6 +44,44 @@ type ChainAddress struct {
 	Approvals []string `ion:"approvals"`
 }
 
+// Vault represents a key which has been broken into shamir shares and is used for encrypting
+// secrets.
+type Vault struct {
+	PublicKey      string   `ion:"publicKey"`
+	Creator        string   `ion:"creator"`
+	Approvals      []string `ion:"approvals"`
+	Threshold      int      `ion:"threshold"`
+	OperatorKeys   []string `ion:"operatorKeys"`
+	IdempotencyKey string   `ion:"idempotencyKey"`
+	// must be unexported. these values should never be presisted to QLDB
+	shares paymentLib.CreateVaultResponse
+}
+
+// SetIdempotencyKey takes and existing vault, checks if it has valid threshold and operators, and
+// generates a UUIDv5 using our Payment Service namespace, threshold, and operators.
+func (v *Vault) SetIdempotencyKey() error {
+	if v.Threshold < 1 {
+		return errors.New("invalid threshold")
+	}
+	if v.OperatorKeys == nil || len(v.OperatorKeys) < 1 || len(v.OperatorKeys) < v.Threshold {
+		return errors.New("invalid number of operator keys")
+	}
+	if len(v.PublicKey) < 1 {
+		return errors.New("public key must be defined before idempotency key can be generated")
+	}
+	// We have to sort the opKeys to ensure that the idempotency key we generate is the same for the
+	// same set of keys.
+	slices.Sort(v.OperatorKeys)
+	v.OperatorKeys = slices.Compact(v.OperatorKeys)
+	// Generate an idempotency key using the keys and threshold. This is used to prevent us from
+	// creating multiple vaults with the same configuration.
+	v.IdempotencyKey = uuid.NewSHA1(
+		uuid.MustParse("3c0e75eb-9150-40b4-a988-a017d115de3c"),
```

Comment:
is this a magic UUID? I've seen the same one used a few places now but it's not clear what it's for.

Reply:
It's the namespace for all of our UUIDv5s. It needs to remain the same value effectively forever. I'll make an issue to consolidate its use and document its meaning.

Reply:
Actually, let's just use the existing issue https://github.com/brave-intl/bat-go/issues/2380